src/main/java/org/jabref/gui/journals/AbbreviateAction.java

```diff
@@ -53,10 +63,20 @@ public void run() {
 
                 return false;
             };
+            tasks.add(callable);
+        }
 
-            boolean result = JabRefExecutorService.INSTANCE.executeAndWait(callable);
-            if (result) {
-                count++;
+        // Execute the callables and wait for the results.
+        List<Future<Boolean>> futures = JabRefExecutorService.INSTANCE.executeAll(tasks);
+
+        // Evaluate the results of the callables.
+        for (Future<Boolean> future : futures) {
```

Comment:
`futures.stream().map(f -> f.get()).countif(...` --> Please try to use Java8's streaming API - if possible. 😇

Reply:
I checked stackoverflow. It should read `count = futures.stream().filter(f -> f.get()).count();` Shouldn't it?

Reply:
Isn't using a filter handier? Like this:
`futures.stream().filter(...).count()`?

Reply:
Indeed. Using a filter seems to be the better option to me. Only downside of this is, we need to catch  the Exceptions within the filter... but ok, I'll do it that way.